code/datums/elements/frozen.dm

```diff
@@ -0,0 +1,47 @@
+///simple element to handle frozen obj's
+/datum/element/frozen
+	element_flags = ELEMENT_DETACH
+
+/datum/element/frozen/Attach(datum/target)
```

Comment:
previously checking if an object was frozen and updating it was done in /turf/open/Entered() but frozen objects are a tiny minority of movables entering turfs. so it makes sense to offload this into an element that does the work we need